### ISSUE TITLE
Unloading a plugin throws NullReferenceException

### ISSUE THREAD

**Issue by oliverbooth** [oliverbooth/BrackeysBot#3]
This seems to happen whether using `plugin unload` as provided by [BrackeysBot.Core](https://github.com/oliverbooth/BrackeysBot.Core), or while exiting the application. This leads me to believe the exception is thrown from within the unloading logic itself, which is defined in this repository.

**Comment by oliverbooth** [comment]
Stack trace:
```
Unhandled exception. System.AggregateException: One or more hosted services failed to stop. (Object reference not set to an instance of an object.)
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at BrackeysBot.Plugins.SimplePluginManager.UnloadPlugin(IPlugin plugin) in C:\Users\Oliver\source\repos\BrackeysBot\BrackeysBot\Plugins\SimplePluginManager.cs:line 309
   at BrackeysBot.BrackeysBotApp.StopAsync(CancellationToken cancellationToken) in C:\Users\Oliver\source\repos\BrackeysBot\BrackeysBot\BrackeysBotApp.cs:line 121
   at Microsoft.Extensions.Hosting.Internal.Host.StopAsync(CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.Extensions.Hosting.Internal.Host.StopAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.WaitForShutdownAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Program.<Main>$(String[] args) in C:\Users\Oliver\source\repos\BrackeysBot\BrackeysBot\Program.cs:line 8
   at Program.<Main>(String[] args)
```

**Comment by oliverbooth** [comment]
Line 309 refers to the unloading of the AssemblyLoadContext.
```cs
monoPlugin.LoadContext.Unload();
```

Since `monoPlugin` is assured not null, this leaves null `LoadContext` as the only cause

**Comment by oliverbooth** [comment]
Fixed with 1cf9f3972513e7c05aa83d4cb5472acb709ccbd4
